ubloxgps/gpsparser.go

```diff
@@ -0,0 +1,137 @@
+package ubloxgps
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+)
+
+type GPSParser struct {
+	gpsDevice GPSDevice
+}
+
+// fix is a GPS location fix
+type Fix struct {
```

Comment:
@deadprogram should this struct be included in the drivers package itself? Meaning "tinygo.org/x/drivers". If it is, it is possible to define a common interface because all "fix" structs are of the same type.

Reply:
Well, I think that it is more likely that we would want to add the UART interface to this package than that we will implement an entirely different driver, just because UBLOX is the main manufacturer out there.

Reply:
Maybe there should be a drivers/gps that implements the basic gps over UART function and then another drivers/ubloxgps that supports both I2C and UART. The only difference btw the I2C and UART impl is the [fillBuffer()](https://github.com/HarringayMakerSpace/drivers/blob/ubloxgps/ubloxgps/ubloxgps.go#L68) and [available()](https://github.com/HarringayMakerSpace/drivers/blob/ubloxgps/ubloxgps/ubloxgps.go#L84). The main reason I haven't done this is I don't have an MCU which Tinygo supports that has multiple UARTs so it was hard to test, guess I could go get a Circuit Express to sort that out. Then another thing I'd like to add to the u-blox impl is some ublox specific functions to send commands to the gps device to do things like disable the COCOM limits, similar to this: https://github.com/HarringayMakerSpace/HAB/blob/master/HABTracker/gps.ino#L84

Reply:
I think we can do such cleanups later when there really is another device, without breaking user code too badly.